sebs/fission/fission.py

```diff
@@ -0,0 +1,373 @@
+import logging
+import os
+import shutil
+import subprocess
+import docker
+import json
+from time import sleep
+from typing import Dict, Tuple, List
+from sebs.faas.storage import PersistentStorage
+from sebs.cache import Cache
+from sebs.config import SeBSConfig
+from sebs.faas.function import Function
+from sebs.faas.system import System
+from sebs.fission.fissionFunction import FissionFunction
+from sebs.benchmark import Benchmark
+from sebs.fission.config import FissionConfig
+from sebs.fission.minio import Minio
+from tools.fission_preparation import (
+    check_if_minikube_installed,
+    run_minikube,
+    check_if_k8s_installed,
+    check_if_helm_installed,
+    stop_minikube,
+)
+
+
+class Fission(System):
+    language_image: str
+    language_builder: str
+    storage: Minio
+    httpTriggerName: str
+    functionName: str
+    packageName: str
+    envName: str
+    shouldCallBuilder: bool = False
+    _config: FissionConfig
+
+    def __init__(
+        self,
+        sebs_config: SeBSConfig,
+        config: FissionConfig,
+        cache_client: Cache,
+        docker_client: docker.client,
+    ):
+        super().__init__(sebs_config, cache_client, docker_client)
+        self._added_functions: List[str] = []
+        self._config = config
+
+    @staticmethod
+    def name():
+        return "fission"
+
+    @property
+    def config(self) -> FissionConfig:
+        return self._config
+
+    @staticmethod
+    def add_port_forwarding(port=5051):
+        podName = (
+            subprocess.run(
+                f"kubectl --namespace fission get pod -l svc=router -o name".split(),
+                stdout=subprocess.PIPE,
+            )
+            .stdout.decode("utf-8")
+            .rstrip()
+        )
+        subprocess.Popen(
+            f"kubectl --namespace fission port-forward {podName} {port}:8888".split(),
+            stderr=subprocess.DEVNULL,
+        )
+
+    def shutdown(self) -> None:
+        if self.config.shouldShutdown:
+            if hasattr(self, "httpTriggerName"):
+                subprocess.run(f"fission httptrigger delete --name {self.httpTriggerName}".split())
+            if hasattr(self, "functionName"):
+                subprocess.run(f"fission fn delete --name {self.functionName}".split())
+            if hasattr(self, "packageName"):
+                subprocess.run(f"fission package delete --name {self.packageName}".split())
+            if hasattr(self, "envName"):
+                subprocess.run(f"fission env delete --name {self.envName}".split())
+            stop_minikube()
+        self.storage.storage_container.kill()
+        logging.info("Minio stopped")
+
+    def get_storage(self, replace_existing: bool = False) -> PersistentStorage:
+        self.storage = Minio(self.docker_client)
+        return self.storage
+
+    def initialize(self, config: Dict[str, str] = None):
+        if config is None:
+            config = {}
+
+        check_if_minikube_installed()
```

Comment:
As above -> we should assume Fission is configured and run by the user. Here, we should only initialize storage connection (see how it work in OpenWhisk).